Makefile

```diff
@@ -144,9 +144,25 @@ run:
 .PHONY: docker-component # Not intended to be used directly
 docker-component: check-component
 	GOOS=linux GOARCH=amd64 $(MAKE) $(COMPONENT)
-	cp ./bin/$(COMPONENT)_linux_amd64 ./cmd/$(COMPONENT)/$(COMPONENT)
-	docker build -t $(COMPONENT) ./cmd/$(COMPONENT)/
-	rm ./cmd/$(COMPONENT)/$(COMPONENT)
+	cp ./bin/$(COMPONENT)_linux_amd64 ./cmd/$(COMPONENT)
+	docker build -t $(COMPONENT) ./cmd/$(COMPONENT)/ --build-arg=TARGETOS=linux --build-arg=TARGETARCH=amd64
```

Comment:
Is it simpler to use buildx for this too so we don't need to define the args?

Reply:
My opinion at this time is to make the multi-architecture build additive and not require it for all contributors. Since 'make docker-otelcontribcol' is pre-existing, I believe it should continue to work without contributors updating their tooling. If contributors want to build multi-arch images, then they'll need to install buildx (doesn't seem common at this time).

Maybe after we have buildx working as expected in the CI/CD pipeline and are okay with requiring contributors to install an experimental tool, then we can fully migrate to buildx. I'm hoping that it will be fully baked into Docker this year, so it will become a natural progression.